fairlearn/metrics/_metric_frame.py

```diff
@@ -272,26 +290,39 @@ def __init__(self,
                  y_true,
                  y_pred,
                  sensitive_features,
-                 control_features: Optional = None,
+                 control_features=None,
                  sample_params: Optional[Union[Dict[str, Any], Dict[str, Dict[str, Any]]]] = None):
         """Read a placeholder comment."""
         check_consistent_length(y_true, y_pred)
+
+        all_data = pd.DataFrame(dtype='object')
```

Comment:
why is this needed?

Reply:
It was to make the object exist as a `DataFrame`. I have changed how it's constructed.